pkg/session/session.go

```diff
@@ -95,22 +127,38 @@ func GetOrCreate(
 	return &session, nil
 }
 
-func newClient(ctx context.Context, url *url.URL, thumprint string) (*govmomi.Client, error) {
+func newClient(ctx GetOrCreateContext, sessionKey string, url *url.URL, thumprint string, feature Feature) (*govmomi.Client, error) {
 	insecure := thumprint == ""
 	soapClient := soap.NewClient(url, insecure)
 	if !insecure {
 		soapClient.SetThumbprint(url.Host, thumprint)
 	}
 
-	vimClient, err := vim25.NewClient(ctx, soapClient)
+	vimClient, err := vim25.NewClient(ctx.context, soapClient)
 	if err != nil {
 		return nil, err
 	}
+
 	c := &govmomi.Client{
 		Client:         vimClient,
 		SessionManager: session.NewManager(vimClient),
 	}
-	if err := c.Login(ctx, url.User); err != nil {
+
+	if feature.EnableKeepAlive {
+		vimClient.RoundTripper = session.KeepAliveHandler(vimClient.RoundTripper, feature.KeepAliveDuration * time.Minute, func(tripper soap.RoundTripper) error {
+			if ok, _ := c.SessionManager.SessionIsActive(ctx.context); ok {
+				return nil
+			}
+			if err := c.Login(ctx.context, url.User); err != nil {
```

Comment:
actually I don't know what is the behavior here. I guess we need to test it out. build a cluster and let it run for 2 days lets see.